libsolidity/formal/CHC.cpp

```diff
@@ -323,6 +376,109 @@ void CHC::unknownFunctionCall(FunctionCall const&)
 	m_unknownFunctionCallSeen = true;
 }
 
+/*
+ * Loop encoding is described as follows.
+ * Let f_i the function body block before the loop.
+ * Create the following blocks:
+ * loop_header, containing constraints created by the condition (potential side effect statements).
+ * loop_body, containing constraints from the loop body.
+ * f_j, the function body block after the loop.
+ * Create the following edges:
+ * f_i -> loop_header
+ * loop_header -> loop_body
+ * loop_header -> f_j
+ * loop_body -> loop_header, if there were no nested loops inside this loop body.
+ * f_k -> loop_header, if there was at least one nested loop inside this loop body,
+ * where f_k is the function body block after the latest nested loop.
+ */
+void CHC::visitLoop(
+	BreakableStatement const& _loop,
+	Expression const* _condition,
+	Statement const& _body,
+	ASTNode const* _postLoop
+)
+{
+	solAssert(m_currentFunction, "");
+	auto const& functionBody = m_currentFunction->body();
+
+	// Create loop header block and edge from function body to loop header.
+	m_predicates[&_loop] = createBlock(sort(functionBody), "loop_" + to_string(_loop.id()));
```

Comment:
This statement together with the next three statements reoccurs again and again in almost the same way right? Would it help to make this a function? The FunctionDefinition body basically has the same as well and so on...